### PR TITLE
[plugins/aws_k8s][feat] Support aws_k8s deferred edge collection

### PR DESCRIPTION
**Pull request by meln1k**
# Description

A new plugin that allows linking k8s and aws eks resources

- [x] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`
- [ ] Document new or updated functionality (someengineering/resoto.com#XXXX)

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: plugins/aws_k8s/resoto_plugin_aws_k8s/__init__.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            graph.add_deferred_edge(
+                BySearchCriteria(f"is(aws_eks_nodegroup) and reported.id={nodegroup}"),
+                ByNodeId(resource.chksum),
+            )
+
+
+def link_k8s_cluster_to_eks_cluster(graph: Graph, resource: BaseResource) -> None:
+    if resource.kind == "kubernetes_cluster":
+        graph.add_deferred_edge(
+            BySearchCriteria(f"is(aws_eks_cluster) and reported.arn={resource.id}"),
```
Comment (aquamatthias) [inline plugins/aws_k8s/resoto_plugin_aws_k8s/__init__.py:29]: `arn`==`id` sounds wrong to me.

Comment (meln1k) [reply plugins/aws_k8s/resoto_plugin_aws_k8s/__init__.py:29]: that's how it is collected.

Comment (aquamatthias) [reply plugins/aws_k8s/resoto_plugin_aws_k8s/__init__.py:29]: Yes this is coming from the k8s side - it does not know about arns. This is fine.

**Review by aquamatthias — APPROVED** [review]
Thanks!

**Review by aquamatthias — APPROVED** [review]
thanks!

**Pull request merged**